File: all-posts.php
<?php

require('admin/load.php');

function get_all_posts() {
	global $tbdb;

	$sql = "SELECT id,date,title,content FROM posts WHERE type='post' AND status='public' ORDER BY date DESC";

	$results = $tbdb->query($sql);

	if($results) {
		while($row = $results->fetch_object()) {
			echo '<li><a target="_blank" href="/',$row->id,'/">',$row->title,'</a></li>',"\n";
		}
	}
}

?><!doctype html>
<html>
<head>
<meta charset="UTF-8" />
<title>所有文章 - <?php echo $tbopt->get('blog_name'); ?></title>
</head>
<body>
<h1>所有文章：</h1>
<div style="padding: 0px 2em;">
<ul>
<?php get_all_posts(); ?>
</ul>
</div>
</body>
</html>
<?php

File: server/all-posts.go
package main

type xAllPostResult struct {
	ID    int64  `json:"id"`
	Title string `json:"title"`
}

type xAllPostResults []xAllPostResult

func getAllPosts(tx Querier) (xAllPostResults, error) {
	var err error
	query := `SELECT id,title FROM posts WHERE type='post' AND status='public' ORDER BY date DESC`
	rows, err := tx.Query(query)
	if err != nil {
		return nil, err
	}

	//defer rows.Close()

	rets := make(xAllPostResults, 0)

	for rows.Next() {
		var ret xAllPostResult
		err = rows.Scan(&ret.ID, &ret.Title)
		if err != nil {
			return nil, err
		}
		rets = append(rets, ret)
	}

	return rets, nil
}
